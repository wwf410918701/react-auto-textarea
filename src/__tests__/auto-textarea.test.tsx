import { render, screen, fireEvent } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { AutoTextArea } from "..";

describe("Auto textarea can correctly ", () => {
  it("render", () => {
    const result = render(<AutoTextArea />);
    expect(result).toMatchSnapshot();
  });

  it("render with short content that not required to span horizontally", () => {
    const result = render(<AutoTextArea value="content" />);
    expect(result).toMatchSnapshot();
  });

  it("expand horizontally", () => {
    const result = render(<AutoTextArea value="content content content" />);
    expect(result).toMatchSnapshot();
  });

  it("expand both vertically and horizontally", () => {
    const result = render(
      <AutoTextArea value="extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content" />,
    );
    expect(result).toMatchSnapshot();
  });

  it("render with styles", () => {
    const result = render(
      <AutoTextArea
        style={{
          border: "1px solid black",
          display: "inline-block",
          minWidth: "50px",
          maxWidth: "400px",
          minHeight: "30px",
          maxHeight: "120px",
          fontSize: "40px",
        }}
        value="extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content"
      />,
    );
    expect(result).toMatchSnapshot();
  });
});

const setup = (label?: string, onChange?: (e: any) => string) => {
  const ariaLabel = label ?? "auto-text-area";
  render(<AutoTextArea aria-label={ariaLabel} onChange={onChange} />);
  const input: HTMLTextAreaElement = screen.getByLabelText(ariaLabel);
  return input;
};

describe("When receive input, auto textarea can", () => {
  it("receive value", () => {
    const textarea = setup("receive value");
    fireEvent.change(textarea, { target: { value: "input" } });
    expect(textarea.value).toBe("input");
  });

  it("update horizontal size correctly", () => {
    const textarea = setup("update horizontal size correctly");
    fireEvent.change(textarea, {
      target: { value: "content content content" },
    });
    expect(textarea).toMatchSnapshot();
  });

  it("update horizontal and vertical size correctly", () => {
    const textarea = setup("update horizontal and vertical size correctly");
    fireEvent.change(textarea, {
      target: {
        value:
          "extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content",
      },
    });
    expect(textarea).toMatchSnapshot();
  });

  it("trigger callback with correct prop", () => {
    const content =
      "extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content";
    const callback = vi.fn((e) => {
      return e.target.value;
    });
    const textarea = setup("trigger callback with correct prop", callback);

    fireEvent.change(textarea, {
      target: {
        value: content,
      },
    });
    expect(callback).toBeCalledTimes(1);
    expect(callback).toReturnWith(content);
  });

  it("trigger reset correctly", () => {
    const emptyContent = "";
    const callback = vi.fn((e) => {
      return e.target.value;
    });
    const textarea = setup("trigger reset correctly", callback);

    fireEvent.change(textarea, {
      target: {
        value: "some input",
      },
    });
    expect(callback).toBeCalledTimes(1);
    fireEvent.change(textarea, {
      target: {
        value: emptyContent,
      },
    });
    expect(callback).toBeCalledTimes(2);
    expect(callback).toReturnWith(emptyContent);
    expect(textarea).toMatchSnapshot();
  });
});
