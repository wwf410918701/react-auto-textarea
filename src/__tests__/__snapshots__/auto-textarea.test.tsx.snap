// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Auto textarea can correctly  > expand both vertically and horizontally 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div />
            <br />
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content content content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content content content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
          </textarea>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="auto-textarea_mockTextareaBorder__ofoscv0"
    >
      <div
        class="auto-width-wrapper_root__6gn7050"
      >
        <div
          class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
        >
          <div>
            extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
          </div>
        </div>
        <textarea
          class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
        >
          extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
        </textarea>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Auto textarea can correctly  > expand horizontally 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div />
            <br />
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content content content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content content content
          </textarea>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="auto-textarea_mockTextareaBorder__ofoscv0"
    >
      <div
        class="auto-width-wrapper_root__6gn7050"
      >
        <div
          class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
        >
          <div>
            content content content
          </div>
        </div>
        <textarea
          class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
        >
          content content content
        </textarea>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Auto textarea can correctly  > render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div />
            <br />
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="auto-textarea_mockTextareaBorder__ofoscv0"
    >
      <div
        class="auto-width-wrapper_root__6gn7050"
      >
        <div
          class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
        >
          <div />
          <br />
        </div>
        <textarea
          class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Auto textarea can correctly  > render with short content that not required to span horizontally 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div />
            <br />
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content
          </textarea>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="auto-textarea_mockTextareaBorder__ofoscv0"
    >
      <div
        class="auto-width-wrapper_root__6gn7050"
      >
        <div
          class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
        >
          <div>
            content
          </div>
        </div>
        <textarea
          class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
        >
          content
        </textarea>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Auto textarea can correctly  > render with styles 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div />
            <br />
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              content content content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            content content content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
          </textarea>
        </div>
      </div>
    </div>
    <div>
      <div
        class="auto-textarea_mockTextareaBorder__ofoscv0"
        style="border: 1px solid black; display: inline-block; min-width: 50px; max-width: 400px; min-height: 30px; max-height: 120px; font-size: 40px;"
      >
        <div
          class="auto-width-wrapper_root__6gn7050"
        >
          <div
            class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
          >
            <div>
              extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
            </div>
          </div>
          <textarea
            class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
          >
            extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
          </textarea>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="auto-textarea_mockTextareaBorder__ofoscv0"
      style="border: 1px solid black; display: inline-block; min-width: 50px; max-width: 400px; min-height: 30px; max-height: 120px; font-size: 40px;"
    >
      <div
        class="auto-width-wrapper_root__6gn7050"
      >
        <div
          class="auto-width-wrapper_widthHolder__6gn7051 shareStyles_fontStyle__dlrh0"
        >
          <div>
            extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
          </div>
        </div>
        <textarea
          class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
        >
          extra long contentextra long contentextra long contentextra long contentextra long contentextra long contentextra long content
        </textarea>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`When receive input, auto textarea can > trigger reset correctly 1`] = `
<textarea
  aria-label="trigger reset correctly"
  class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
/>
`;

exports[`When receive input, auto textarea can > update horizontal and vertical size correctly 1`] = `
<textarea
  aria-label="update horizontal and vertical size correctly"
  class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
/>
`;

exports[`When receive input, auto textarea can > update horizontal size correctly 1`] = `
<textarea
  aria-label="update horizontal size correctly"
  class="input_root__1wnzxyg0 shareStyles_fontStyle__dlrh0"
/>
`;
